application/lib/console.dart

```diff
@@ -111,7 +119,45 @@ class _ConsoleState extends State<Console> {
               _buildChip('take all'),
               _buildChip('drop all'),
               _buildChip('debug status'),
-              _buildChip('debug locations'),
+              OutlinedButton(
+                child: const Text('Teleport'),
+                onPressed: () {
+                  widget.game
+                      .sendMessage('debug locations')
+                      .then((String result) {
+                    final Iterable<String> locations = result
+                        .split('\n')
+                        .skip(1)
+                        .takeWhile((String line) => line.isNotEmpty)
+                        .map((String line) => line.substring(3));
+                    showDialog(
+                      context: context,
+                      builder: (BuildContext context) {
+                        return BoilerplateDialog(
+                          title: 'Teleport to:',
+                          children: locations
+                              .map(
+                                (String location) => ActionChip(
+                                  label: Text(location),
+                                  onPressed: () {
```

Comment:
ideally you'd return the location from the dialog here, then use the return value of showDialog to call the server